src/storage.rs

```diff
@@ -13,6 +14,30 @@ use super::errors::Error;
 pub type DatabaseConnection = r2d2::PooledConnection<SqliteConnectionManager>;
 pub type DatabaseConnectionPool = r2d2::Pool<SqliteConnectionManager>;
 
+#[derive(PartialEq, Eq, Hash)]
+pub struct RoomId {
+    id: String,
+}
+
+lazy_static::lazy_static! {
+    // Alphanumeric, Decimals and "-" only
+    static ref REGULAR_CHARACTERS_ONLY: Regex = Regex::new(r"^[a-zA-Z0-9-]*$").unwrap();
```

Comment:
This might need to be broadened. Only allows alphanumeric or "-"

Reply:
I wonder if we are introducing incompatibilities with some of the existing groups with this.

Reply:
Yeah have shot a few messages to the session team to get their thoughts on this

Reply:
this is probably checked somewhere else, but looking at this we allow an empty roomid. Do we want to force a lower limit length? Or just not empty? Also, do we want to limit the upper limit length to like 64 chars?

We can probably safely allow _ too

Reply:
updated to include underscores, upper limit of 64 characters and lower limit of 1 character

Reply:
Do we want room names to be case-insensitive? Maybe only allow lower-case characters?